packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1482 @@
+import 'dart:io' as io;
+
+import 'package:http_parser/http_parser.dart';
+import 'package:relic/src/headers/custom/custom_headers.dart';
+import 'package:relic/src/headers/extension/string_list_extensions.dart';
+import 'package:relic/src/headers/parser/headers_parser.dart';
+import 'package:relic/src/headers/parser/common_types_parser.dart';
+import 'package:relic/src/method/method.dart';
+
+import 'typed/typed_headers.dart';
+
+import '../body/body.dart';
+
+sealed class Headers {
```

Comment:
Suggestion: Use `base` class.

We don't want exhaustive search for the classes defined from this, we simply want to prohibit implmenetations of this class outside of the library. I think `base` is the right choice in that case (since we need to inherit from this class for the copyWith method)

Reply:
You can read more here: https://dart.dev/language/class-modifiers-for-apis#the-base-modifier